### PR TITLE
Taunt Implementation

### PR DESCRIPTION
**Pull request by mKroogz**
## What are the changes?
This PR implements Taunt and Taunt handling logic around currently implemented use/edge cases.

## Why am I doing these changes?
I wanted a fun first PR to contribute to this awesome game and a friend asked me to implement Taunt.  

## What did change?
I have added implemented the Taunt move with logic alongside the conditions that that would cause it fail.  I've also added the (english) dialogue for all of the taunt messages and interactions.

### Screenshots/Videos
Will try to add these shortly. 

## How to test the changes?
* Run locally and start a new game with all Sinnoh starters
* Exclusively take lures to enhance chance of double battles
* Level Chimchar to 9 and learn Taunt
* Taunt will work on enemies but in double battle you can taunt your ally and test a lot better
* Try to use status moves and they should fail
* You can also test that Taunt leaves on swap

## Checklist
- [ ] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/system/pokemon-data.ts [hunk 1]
```diff
@@ -120,6 +120,8 @@ export default class PokemonData {
         this.summonData.moveQueue = source.summonData.moveQueue;
         this.summonData.disabledMove = source.summonData.disabledMove;
         this.summonData.disabledTurns = source.summonData.disabledTurns;
+        this.summonData.isTaunted = source.summonData.isTaunted;
+        this.summonData.tauntedTurns = source.summonData.tauntedTurns;
```
Comment (OrangeRed) [inline src/system/pokemon-data.ts:124]: If you're going to be keeping track of turns like this then Taunt might be better implemented as a BattleTag or StatusEffect

Comment (mKroogz) [reply src/system/pokemon-data.ts:124]: Ooooooh ok!  I'll Iook into it right away :)

Comment (mKroogz) [reply src/system/pokemon-data.ts:124]: I had used Disable as a guideline since it was similar :)

**Comment by zacharied** [comment]
See #2051, #2159.

**Comment by mKroogz** [comment]
> See #2051, #2159.

Should I wait for [#2051](https://github.com/pagefaultgames/pokerogue/pull/2051) and adapt my branch around it?

**Comment by zacharied** [comment]
@mKroogz, sorry, I updated that comment, check it again. I'm not entirely sure what should be done about this.

**Comment by mKroogz** [comment]
> @mKroogz, sorry, I updated that comment, check it again. I'm not entirely sure what should be done about this.

I read updated comment, unchecked,  and now I'm also unsure 🤔 

A lot of my logic was based on how disable was handled especially in the isUsable method and just made a change in approach for what moves can't be used.

**Comment by zacharied** [comment]
I'd say feel free to leave it as an example of an alternate approach, and a dev can choose which to merge. I don't know if my solution will get merged in the end either.

File: src/data/ability.ts [hunk 2]
```diff
@@ -1516,6 +1516,21 @@ export class IgnoreOpponentEvasionAbAttr extends AbAttr {
...
+    super(false);
+  }
+
+  apply(pokemon: Pokemon, passive: boolean, cancelled: Utils.BooleanHolder, args: any[]): boolean {
+    cancelled.value = true;
+    return true;
+  }
+
+  getTriggerMessage(pokemon: Pokemon, abilityName: string, ...args: any[]): string {
+    return getPokemonMessage(pokemon, `'s ${abilityName} prevented it from being Taunted!`);
```
Comment (CodeTappert) [inline src/data/ability.ts:1530]: use getPokemonWithAffix here please and make this localizable obviously

File: src/data/move.ts [hunk 3]
```diff
@@ -3767,6 +3767,25 @@ export class DisableMoveAttr extends MoveEffectAttr {
...
+
+  apply(user: Pokemon, target: Pokemon, move: Move, args: any[]): boolean {
+    if (!super.apply(user, target, move, args)) {
+      return false;
+    }
+
+      target.summonData.isTaunted = true;
+      target.summonData.tauntedTurns = 3;
+
+      user.scene.queueMessage(getPokemonMessage(target, ` fell for the taunt!`));
```
Comment (CodeTappert) [inline src/data/move.ts:3783]: needs to be localizable

File: src/phases.ts [hunk 4]
```diff
@@ -2549,6 +2555,9 @@ export class MovePhase extends BattlePhase {
       if (this.move.moveId && this.pokemon.summonData?.disabledMove === this.move.moveId) {
         this.scene.queueMessage(`${this.move.getName()} is disabled!`);
       }
+      else if (this.move.moveId && this.pokemon.summonData?.isTaunted) {
+        this.scene.queueMessage(`${this.pokemon.name} can't use\n${this.move.getName()} after the taunt!`);
```
Comment (CodeTappert) [inline src/phases.ts:2559]: localizable please

**Comment by torranx** [comment]
Closing this in favor of https://github.com/pagefaultgames/pokerogue/pull/2051

**Pull request closed without merging**